Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm

```diff
@@ -360,7 +360,11 @@ + (WKWebsiteDataStore *)dataStoreForIdentifier:(NSUUID *)identifier
     if (!identifier)
         return nil;
 
-    return wrapper(WebKit::WebsiteDataStore::dataStoreForIdentifier(UUID(identifier)));
+    auto uuid = UUID(identifier);
+    if (!uuid)
+        return nil;
```

Comment:
The return type is not nullable.  This should instead raise an NSInvalidArgumentException.  Same with the check on line 360.
Also, we should use isValidKey instead to also check for the hash table deleted value and make a test that hits that if possible.  We should document that the NSUUID cannot be the empty UUID in the header.

Reply:
Updated to make it throw error.